src/disrupt.sh

```diff
@@ -23,15 +23,25 @@ if [[ ! -d "${IMG_ASSET_DIR}" || ! $(find "${IMG_ASSET_DIR}" -mindepth 1 -iname
     DICE_ROLL=0
 fi
 
-# add the image if necessary
-if [ "${DICE_ROLL}" -eq 1 ]; then
-    RANDOM_IMAGE=$(find "${IMG_ASSET_DIR}" -type f | shuf -n 1)
+#Check for an image associated with the fact
+fact_cmd="$RUN_DIR/src/lf_imap.py $RANDOM_FACT"
+ASSOCIATED_IMAGE=$($fact_cmd)
+DICE_ROLL=${DICE_ROLL/.*}
 
-    cat "${MSG_TEMPLATE_IMG}" >> "${TEMP_FILE}"
-    base64 "${RANDOM_IMAGE}" >> "${TEMP_FILE}"
+if [ "${DICE_ROLL}" -eq 1 ]; then
+	if [ "$ASSOCIATED_IMAGE" ] && [ -s "$ASSOCIATED_IMAGE" ]; then
+		echo "Using associated image"
+		RANDOM_IMAGE=$ASSOCIATED_IMAGE
+	else
+		# add the image if no matching but random chance said to
+		echo "Using random image"
+		RANDOM_IMAGE=$(find "${IMG_ASSET_DIR}" -type f | gshuf -n 1)
+	fi
+	cat "${MSG_TEMPLATE_IMG}" >> "${TEMP_FILE}"
+	base64 "${RANDOM_IMAGE}" >> "${TEMP_FILE}"
 else
-    # otherwise just end the body
-    echo '--body--' >> "${TEMP_FILE}"
+	# otherwise just end the body
```

Comment:
I think you have a tab here instead of 4 spaces 😜

Reply:
tabs >>>>>> spaces